File: RelationExample/RelationExample/Computer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RelationExample
{
    class Computer
    {
        private CPU cpu;
        private GraphicCard graphicCard;

        public Computer(CPU cpu, GraphicCard graphicCard)
        {
            this.cpu = cpu;
            this.graphicCard = graphicCard;
        }

        public void PrintCPUCores()
        {
            Console.WriteLine("The CPU of the computer has {0} cores", cpu.NumberOfCores);
        }
        public void PrintDRam()
        {
            Console.WriteLine("The graphic card of the computer has {0} gb dRam" , graphicCard.DRam);
        }
    }
}

File: RelationExampleC++/RelationExampleC++/Computer.cpp
#include "Computer.h"


Computer::Computer(CPU cpu, GraphicCard graphicCard) : m_cpu(cpu), m_graphicCard(graphicCard)
{}

void Computer::PrintCPUCores()
{
	std::cout << "The CPU of the computer has " << m_cpu.GetNumberOfCores() << " cores \n";
}

void Computer::PrintDRam()
{
	std::cout << "The graphic card of the computer has " << m_graphicCard.GetDRam() << " gb dRam \n";
}